sentinelsat/sentinel.py

```diff
@@ -318,28 +315,42 @@ def download(self, id, directory_path='.', checksum=False, check_existing=False)
 
         self.logger.info('Downloading %s to %s' % (id, path))
 
-        # Check if the file exists and passes md5 test
-        # The download function will by default continue the download if the file exists but is incomplete
-        if exists(path) and getsize(path) == product_info['size']:
-            if not check_existing or _md5_compare(path, product_info['md5']):
-                self.logger.info('%s was already downloaded.' % path)
-                return product_info
+        if exists(path):
```

Comment:
I felt that it is a bit orthogonal to the main functionality of `download()`. Though, I agree, I have found this functionality useful quite often myself, too. Since the `.zip` files can now be expected to be complete and verified with this PR's changes, there is normally no need to recheck existing files.
There are a couple of options:
* Add a separate method for this (`checksum()`?) functionality, which takes a list of `id`s as input.
* Not implement this at all. The zip files can be verified quite easily with `zip -T` or `7z t` commands or via a GUI.
* Re-add it to `download()`. Since you are likely interested in fixing the broken files rather than just deleting them, `download()` is actually a quite sensible place for this functionality.

I guess `download()` is still the best place for this feature. :)